src/goTranspiler.ts

```diff
@@ -710,41 +744,104 @@ ${this.getIden(identation)}PanicOnError(${varName})`;
 
         const op = node.operatorToken.kind;
 
-        if (left.kind === ts.SyntaxKind.TypeOfExpression) {
-            const typeOfExpression = this.handleTypeOfInsideBinaryExpression(node, identation);
-            if (typeOfExpression) {
-                return typeOfExpression;
-            }
+        // ---------------------------------------------------------------
+        // Array destructuring assignment:  [a, b] = foo()
+        // Transforms into:
+        // __tmpX := foo()
+        // a = GetValue(__tmpX, 0)
+        // b = GetValue(__tmpX, 1)
+        // ---------------------------------------------------------------
+        if (op === ts.SyntaxKind.EqualsToken &&
+            left.kind === ts.SyntaxKind.ArrayLiteralExpression) {
+            const elems = (left.elements as any[]);
+            const tmpVar = `__tmp${this.getRandomNameSuffix()}`;
```

Comment:
We shouldn't use this ( I used in the past but removed it) because everytime we build a new version the produced code will be different and that's a "problem" in the pipeline because we're always pushing "new" code that it's in fact the exact same, we should use instead         const returnRandName = "retRes" + this.getLineBasedSuffix(node);